From f5fa1596354980975c521bbcdc09154adc4fd021 Mon Sep 17 00:00:00 2001
From: Benjamin C Meyer <benjamin.meyer@torchmobile.com>
Date: Fri, 27 Mar 2009 13:55:29 -0400
Subject: [PATCH] [PATCH] Update the page actions when a finishing loading even if the frame is
 not the top frame such as when browsing inside of a website with frames.

---
 .../qt/WebCoreSupport/FrameLoaderClientQt.cpp      |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/3rdparty/webkit/WebKit/qt/WebCoreSupport/FrameLoaderClientQt.cpp b/src/3rdparty/webkit/WebKit/qt/WebCoreSupport/FrameLoaderClientQt.cpp
index c421d42..fb7acf8 100644
--- a/src/3rdparty/webkit/WebKit/qt/WebCoreSupport/FrameLoaderClientQt.cpp
+++ b/src/3rdparty/webkit/WebKit/qt/WebCoreSupport/FrameLoaderClientQt.cpp
@@ -382,7 +382,7 @@ void FrameLoaderClientQt::dispatchDidFinishLoad()
 
     m_loadSucceeded = true;
 
-    if (m_frame->tree()->parent() || !m_webFrame)
+    if (!m_webFrame)
         return;
     m_webFrame->page()->d->updateNavigationActions();
 }
-- 
1.6.0.4

